pkg/webhook/webhook_test.go

```diff
@@ -94,17 +97,32 @@ func TestEnsureWebhookConfigurations(t *testing.T) {
 		}
 
 		cl := fake.NewClientBuilder().Build()
+		var validatingWhCfg admissionregistrationv1.ValidatingWebhookConfiguration
+		var mutatingWhCfg admissionregistrationv1.MutatingWebhookConfiguration
 		// prove webhook configurations don't exist
-		require.True(t, k8serrors.IsNotFound(cl.Get(context.Background(), types.NamespacedName{Name: c.validatingWebhookConfigName}, &admissionregistrationv1.ValidatingWebhookConfiguration{})), "expected not to find validating webhook config")
-		require.True(t, k8serrors.IsNotFound(cl.Get(context.Background(), types.NamespacedName{Name: c.mutatingWebhookConfigName}, &admissionregistrationv1.MutatingWebhookConfiguration{})), "expected not to find mutating webhook config")
+		require.True(t, k8serrors.IsNotFound(cl.Get(context.Background(), types.NamespacedName{Name: c.validatingWebhookConfigName}, &validatingWhCfg)), "expected not to find validating webhook config")
+		require.True(t, k8serrors.IsNotFound(cl.Get(context.Background(), types.NamespacedName{Name: c.mutatingWebhookConfigName}, &mutatingWhCfg)), "expected not to find mutating webhook config")
 
 		// prove webhook configurations exist after ensuring
-		require.NoError(t, c.EnsureWebhookConfigurations(context.Background(), cl), "unexpected error")
-		require.NoError(t, cl.Get(context.Background(), types.NamespacedName{Name: c.validatingWebhookConfigName}, &admissionregistrationv1.ValidatingWebhookConfiguration{}), "unexpected error getting validating webhook config")
-		require.NoError(t, cl.Get(context.Background(), types.NamespacedName{Name: c.mutatingWebhookConfigName}, &admissionregistrationv1.MutatingWebhookConfiguration{}), "unexpected error getting mutating webhook config")
+		require.NoError(t, c.EnsureWebhookConfigurations(context.Background(), cl, globalCfg), "unexpected error")
+		require.NoError(t, cl.Get(context.Background(), types.NamespacedName{Name: c.validatingWebhookConfigName}, &validatingWhCfg), "unexpected error getting validating webhook config")
+		require.NoError(t, cl.Get(context.Background(), types.NamespacedName{Name: c.mutatingWebhookConfigName}, &mutatingWhCfg), "unexpected error getting mutating webhook config")
+
+		// prove ownerReferences are set on webhook configurations
```

Comment:
maybe also add a unit test for deleting the namespace which will then delete the whc configurations?

Reply:
Sure!

Reply:
not 100% sure this works with the fake client since it's a garbage collection behavior. If it doesn't no need to agonize over it

Reply:
the unit test to prove the garbage collection is not working as expected since the fake client is not setting any UID to the namespace when its created.